### PR TITLE
Add New Temperature Sensor interface.

### PR DESCRIPTION
**Pull request by ChuefengVang**
Removed method "GetTemperatureBytes()" as it is not needed anymore for new temperature sensor device.

### CODE REVIEW

**Comment by kammce** [comment]
Split this into two PRs. One for just the interface. And the other for the LPC40xx implementation.

**Comment by ChuefengVang** [comment]
Done.

File: firmware/library/L2_HAL/temperature.hpp [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+#include <cstdint>
+#include <cstdio>
+#include "L1_Drivers/i2c.hpp"
+
+class TemperatureInterface
+{
+ public:
+  virtual bool Initialize()                       = 0;
+  virtual int GetCelsius()                        = 0;
```
Comment (kammce) [inline firmware/library/L2_HAL/temperature.hpp]: What size int are you using for this? Please use a specific integer size for the values here. Please see <cstdint> for more details.

File: firmware/library/L2_HAL/temperature.hpp [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <cstdint>
+#include <cstdio>
+#include "L1_Drivers/i2c.hpp"
```
Comment (kammce) [inline firmware/library/L2_HAL/temperature.hpp]: You aren't using any of these includes. Including them will only slow down compilation time. Do not include what are not using. Always include what you are using.

With my comment below, you should keep <cstdint> though.

File: firmware/library/L2_HAL/temperature.hpp [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+class TemperatureInterface
+{
+ public:
+  virtual bool Initialize()                       = 0;
+  virtual int GetCelsius()                        = 0;
+  virtual int GetFahrenheit(int temperature_data) = 0;
```
Comment (kammce) [inline firmware/library/L2_HAL/temperature.hpp]:
```suggestion
  virtual int GetFahrenheit() = 0;
```

Comment (ChuefengVang) [reply firmware/library/L2_HAL/temperature.hpp]: Did all the changes and moved the file over to L2_HAL/sensors/temperature.  Also removed the example main.cpp but couldn't remove the folder. Should be fine there I'll include the example after implementation is pull.

**Pull request merged**